logviewer/index.cgi

```diff
@@ -114,29 +112,51 @@ foreach $e (&extra_log_files()) {
 		push(@cols, &text('index_cmd',
 			"<tt>".&html_escape($e->{'cmd'})."</tt>"));
 		}
-	push(@cols, &html_escape($e->{'desc'}));
+	push(@cols, $e->{'desc'} ? &html_escape($e->{'desc'}) : "");
 	push(@cols, &ui_link("view_log.cgi?extra=".&urlize($e->{'file'} || $e->{'cmd'})."&view=1", $text{'index_view'}) );
+	@cols = sort { $a->[2] cmp $b->[2] } @cols;
 	push(@col3, \@cols);
 	}
 
 # Print sorted table with logs files and commands
-my @acols = (@col1, @col2, @col3);
+my @acols = (@col0, @col1, @col2, @col3);
+
+my $print_header = sub {
+	# Print the header
+	&ui_print_header($text{'index_subtitle'}, $text{'index_title'}, "", undef, 1, 1, 0,
+	&help_search_link("systemd-journal journalctl", "man", "doc"));
+	};
+
+# If no logs are available just show the message
+if (!@acols) {
+	$print_header->();
+	&ui_print_endpage($text{'index_elogs'});
+	}
+
+# If we jump directly to logs just redirect
+if ($config{'skip_index'} == 1) {
+	my $link;
+	$link = $acols[0]->[2];
+	$link =~ s/.*?href\s*=\s*(["']?)(?<link>[^"'\s>]+).*\1?/$+{link}/g;
```

Comment:
Extracting the URL from the HTML seems like a bad idea! It would completely fail if the format of the link ever changes..

Reply:
Format of the link will never change and it will either be `<a href=link>link</a>` or `<a href='link'>link</a>` or `<a href="link">link</a>` ..

Reply:
Regardless, it seems like bad practice to convert something into HTML and then convert it back again. Instead why not just store the URL in a separate array?

Reply:
I can do that ..

Reply:
> Regardless, it seems like bad practice to convert something into HTML and then convert it back again. Instead why not just store the URL in a separate array?

Fixed!